src/Sql/SqlMysql.php

```diff
@@ -107,7 +108,21 @@ public function createdbSql($dbname, $quoted = false)
             $user = sprintf("'%s'@'%s'", $dbSpec['username'], $domain);
             $sql[] = sprintf("DROP USER IF EXISTS %s;", $user);
             $sql[] = sprintf("CREATE USER %s IDENTIFIED WITH mysql_native_password;", $user);
-            $sql[] = sprintf("SET PASSWORD FOR %s = PASSWORD('%s');", $user, $dbSpec['password']);
+
+            // For MariaDB, ALTER USER was introduced in 10.2 so before that use SET PASSWORD.
+            // @see Drupal\Core\Database\Driver\mysql\Connection::getMariaDbVersionMatch()
+            // in Drupal 9.
+            $this->su();
+            $this->alwaysQuery('SELECT VERSION();');
+            $version = trim($this->getProcess()->getOutput());
+            $regex = '/^(?:5\.5\.5-)?(\d+\.\d+\.\d+).*-mariadb.*/i';
+            preg_match($regex, $version, $matches);
+            if (!empty($matches[1]) && version_compare($matches[1], '10.2') < 0) {
```

Comment:
Lets use the 3rd argument of version_compare(). See https://www.php.net/version_compare. Untested suggestion:
```suggestion
            if (!empty($matches[1]) && version_compare($matches[1], '10.2', '<')) {
```